File: tests/csharp/tests-csharp-nunit-allure3-selenium/config/TestConfig.cs
namespace Config;

/// <summary>Java TestConfig analog (UI + in-cell RestSharp).</summary>
public sealed class TestConfig
{
    public string Stand { get; init; } = "prod";
    public string BaseUrl { get; init; } = "";
    public string ApiBaseUrl { get; init; } = "";
    public string ApiHealthService { get; init; } = "backend-java-spring";
    public string WelcomeUsername { get; init; } = "user1";
    public bool EnableAllureRestAssuredListener { get; init; }
    public string Browser { get; init; } = "chrome";
    public string BrowserVersion { get; init; } = "148";
    public string BrowserSize { get; init; } = "1920x1280";
    public bool Headless { get; init; }
    public bool CloseBrowserAfterEach { get; init; } = true;
    public bool CloseBrowserAfterAll { get; init; } = true;
    public bool SkipBlankOpen { get; init; }
    public string RemoteUrl { get; init; } = "";
    public bool EnableVnc { get; init; }
    public bool EnableVideo { get; init; }
    public bool EnableHar { get; init; }
    public string VideoFolder { get; init; } = "";
    public bool AttachBrowserConsoleLogs { get; init; }
    public bool AttachHarLogs { get; init; }
    public bool AttachLastScreenshot { get; init; }
    public bool AttachPageSource { get; init; }
    public bool AttachVideo { get; init; }
    public bool UpdateScreenshots { get; init; }
    public string ScreenshotsDir { get; init; } = "screenshots";
    public double ScreenshotDiffThreshold { get; init; } = 0.015;
}

File: tests/java/tests-java-junit5-api_request-playwright/src/test/java/config/TestConfig.java
package config;

import org.aeonbits.owner.Config;

@Config.LoadPolicy(Config.LoadType.MERGE)
@Config.Sources({
        "system:properties",
        "classpath:config/${env}.properties",
        "classpath:config/default.properties",
})
public interface TestConfig extends Config {

    @DefaultValue("allure3")
    String allureReportMode();

    @DefaultValue("none")
    String allureAgentMode();

    @DefaultValue("false")
    boolean attachBrowserConsoleLogs();

    @DefaultValue("false")
    boolean attachHarLogs();

    @DefaultValue("false")
    boolean attachLastScreenshot();

    @DefaultValue("false")
    boolean attachPageSource();

    @DefaultValue("false")
    boolean attachVideo();

    @DefaultValue("false")
    boolean enableAllureRestAssuredListener();

    @DefaultValue("default")
    String allureRestAssuredListenerStyle();

    @DefaultValue("")
    String baseUrl();

    @DefaultValue("")
    String apiBaseUrl();

    @DefaultValue("backend-java-spring")
    String apiHealthService();

    /** Display name in the home welcome panel after seed login. Mock stand stubs {@code mock-user}. */
    @DefaultValue("user1")
    String welcomeUsername();

    @DefaultValue("")
    String remoteUrl();

    /**
     * This cell is Chromium-only: local Chrome for Testing, or Selenoid
     * {@code wss://…/playwright-chromium/…}. {@code firefox} is not a runtime path.
     */
    @DefaultValue("chrome")
    String browser();

    @DefaultValue("148")
    String browserVersion();

    @DefaultValue("1920x1280")
    String browserSize();

    @DefaultValue("false")
    boolean headless();

    @DefaultValue("false")
    boolean enableHar();

    @DefaultValue("false")
    boolean enableVnc();

    @DefaultValue("false")
    boolean enableVideo();

    /**
     * Public Selenoid video base ({@code https://selenoid.qa.guru/video/}). Hub Allure
     * player uses {@code videoFolder} + the {@code videoName} sent on WS connect.
     */
    @DefaultValue("")
    String videoFolder();

    @DefaultValue("false")
    boolean updateScreenshots();

    @DefaultValue("screenshots")
    String screenshotsDir();

    @DefaultValue("0.015")
    double screenshotDiffThreshold();

    @DefaultValue("true")
    boolean logToConsole();

    @DefaultValue("info")
    String rootLogLevel();

}
